src/System.Threading.Tasks.Extensions/src/CompatibilitySuppressions.xml

```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
+<Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Suppression>
+    <DiagnosticId>CP0008</DiagnosticId>
+    <Target>T:System.Runtime.CompilerServices.AsyncMethodBuilderAttribute</Target>
+    <Left>lib/net461/System.Threading.Tasks.Extensions.dll</Left>
+    <Right>lib/netstandard2.0/System.Threading.Tasks.Extensions.dll</Right>
+    <IsBaselineSuppression>true</IsBaselineSuppression>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>PKV006</DiagnosticId>
+    <Target>.NETPortable,Version=v0.0,Profile=Profile259</Target>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>PKV006</DiagnosticId>
+    <Target>.NETStandard,Version=v1.0</Target>
+  </Suppression>
```

Comment:
Same feedback as in the DispatchProxy PR. Add these TFMs to the package baseline validation ignore list: https://github.com/dotnet/maintenance-packages/blob/92eba09e9ec1004fe6d3db4f4e73f258b5d02579/eng/packaging.targets#L11

Reply:
I want to send it in a separate PR so that I can then merge the commit to all the open PRs.

I am unable to find an example of adding .NET Portable to the baseline suppresion list. I only found these values: https://github.com/search?q=org%3Adotnet+PackageValidationBaselineFrameworkToIgnore+path%3A*.targets&type=code

Do I use `netportable`? Or is it maybe `netportable0.0`?

Reply:
The error message for .NET Portable says:
```
C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): er
ror PKV006: Target framework .NETPortable,Version=v0.0,Profile=Profile259 is no longer supported in the latest version. [D:\maintenance-packages\src\Sys
tem.Threading.Tasks.Extensions\src\System.Threading.Tasks.Extensions.csproj]
```

I couldn't find it among the supported frameworks in the nuget page of the package: https://www.nuget.org/packages/System.Threading.Tasks.Extensions#supportedframeworks-body-tab

I found a Stephen Cleary blog post showing the list of profiles and their values: https://portablelibraryprofiles.stephencleary.com/

My error says profile259, then the value should be `portable-net45+win8+wpa81+wp8`... but there's a problem: in my error it says version 0.0, while in the table it says .NET 4.5.

So it doesn't work if I try to ignore either `portable-net45+win8+wpa81+wp8` or just `portable-net45`.

What am I missing?

Reply:
We want to ignore all portable TFMs. This is very similar to SBRP which uses GenAPI to generate compiled source from prevoius packages. It excludes out-of-support TFMs: https://github.com/dotnet/source-build-reference-packages/blob/69c74e36b3a96556db234bdbbc8e31b5c549bf80/src/packageSourceGenerator/PackageSourceGenerator.proj#L11

We more or less want the same list here (except for net4*). Unfortunately, APICompat doesn't support glob patterns in the exclusion list today. It might make sense to support that. We already do string matching instead of using NuGet.Parse. We could just copy the TFM regex filter code that I wrote for SBRP and use it in PackageValidation: https://github.com/dotnet/source-build-reference-packages/blob/main/src/packageSourceGenerator/PackageSourceGeneratorTask/TargetFrameworkRegexFilter.cs

@ericstj thoughts on this?

Reply:
I submitted https://github.com/dotnet/sdk/pull/38374 as I think this beneficial to others as well.